Global Exception Handler + tests for user_service

ApprehensiveBug opened the pull request.

ApprehensiveBug committed:
Global Exception Handler + test for user_service

ApprehensiveBug committed:
Внёс правки по check_style

AlexeyAntipin commented on src/main/java/school/faang/user_service/controller/handler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,97 @@
+        String errors = e.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map((fieldError) -> fieldError.getField() + " - " + fieldError.getDefaultMessage())
```
1) fieldError не обязательно выносить в скобки, получается избыточность
2) Лучше использовать String.format

AlexeyAntipin commented on src/main/java/school/faang/user_service/controller/handler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private String safeMethod(HttpServletRequest req) {
+        return req != null ? req.getMethod() : "N/A";
```
Может лучше через Optional?

AlexeyAntipin commented on src/main/java/school/faang/user_service/controller/handler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,97 @@
+            IllegalStateException.class
+    })
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleBadInput(RuntimeException e, HttpServletRequest req) {
```
Логика такая же, как и у DataValidationException, может объединить?

ApprehensiveBug committed:
1. Объединил DataValidationException с IllegalArgument / State Exceptions
2. Теперь преобразование каждого fieldError в имя -> сообщение происходит через String.format(), а не ужасную конкатенацию
3. safeMethod() и safeUri() теперь проверяют через Optional.
